CorsixTH/Lua/dialogs/resizables/file_browser.lua

```diff
@@ -197,23 +197,46 @@ function UIFileBrowser:UIFileBrowser(ui, mode, title, vertical_size, root, show_
   -- Initialize the tree control
   self.control = FilteredTreeControl(root, 5, 35, h_size - 10, vertical_size, self.col_bg, self.col_scrollbar, true, show_dates)
     :setSelectCallback(--[[persistable:file_browser_select_callback]] function(node)
-      if node.is_valid_file and (lfs.attributes(node.path, "mode") ~= "directory") then
-        self:choiceMade(node.path)
-      end
+      if self:checkChoice(node) then self:choiceMade(node.path) end
+    end)
+    :setTextboxCallback(--[[persistable:file_browser_textbox_callback]] function(node, label)
+      if self:checkChoice(node) then self:updateTextbox(label) end
     end)
   self:addWindow(self.control)
 
-  -- Create the back button.
-  self:addBevelPanel((h_size - 160) / 2, 340, 160, 30, self.col_bg):setLabel(_S.menu_list_window.back)
-    :makeButton(0, 0, 160, 40, nil, self.buttonBack):setTooltip(_S.tooltip.menu_list_window.back)
+  -- Create the back and ok buttons.
+  local button_size = 135
+  local indent = math.floor((h_size - (2*button_size))/3)
+  self:addBevelPanel(indent, 340, button_size, 30, self.col_bg):setLabel(_S.menu_list_window.back)
+    :makeButton(0, 0, button_size, 40, nil, self.buttonBack):setTooltip(_S.tooltip.menu_list_window.back)
+
+  self:addBevelPanel(h_size - button_size - indent, 340, button_size, 30,
+  self.col_bg):setLabel(_S.menu_list_window.ok)
+    :makeButton(0, 0, button_size, 40, nil, (--[[persistable:filebrowser_ok_callback]] function()
+      if self.confirmName then
+        self:confirmName()
```

Comment:
Had to leave this one behind as is

Reply:
It seems 2 derived classes have `confirmName`?

Technically, if you stick the "elseif" part in the file browser base class, that gets executed for all other browsers. That seems equivalent to this solution to me, at least if it's feasible/desirable to move that code to a separate function.

Or am I missing something?